src/CommandSystem/IdChecker.cpp

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This code has been produced by the European Spallation Source
+// and its partner institutes under the BSD 2 Clause License.
+//
+// See LICENSE.md at the top level for license information.
+//
+// Screaming Udder!                              https://esss.se
+
+#include "IdChecker.h"
+#include <fmt/format.h>
+#include <regex>
+
+namespace Command {
+static const int CommandTimeLimit{10};
+
+constexpr uint32_t MOD_ADLER = 65521;
+
+uint32_t adler32(std::string const &Input) {
+  uint32_t a = 1, b = 0;
+  for (size_t i = 0; i < Input.size(); ++i) {
+    a = (a + Input[i]) % MOD_ADLER;
+    b = (b + a) % MOD_ADLER;
+  }
+  return (b << 16) | a;
+}
+
+std::string const JobIdNoMatchString{
+    "Job-Id did not match the reg.-ex. (id was: \"{:s}\""};
+std::string const JobIdCheckSumWrong{
+    "Job-Id checksum was wrong. It was: 0x{:s}, it should be 0x{:08X}"};
+std::string const JobIdTimestampBad{"Job-Id creation time is bad."};
+std::string const JobIdIdentifierKnown{
+    "Job-Id (\"{}\")has already been used and is therefore rejected."};
```

Comment:
What advantage do you think enforcing a particular format for job id has?
I can't think of any reason worth the extra code and brittleness .

Reply:
> Prevent commands being processed by the wrong file-writer instance.

How? There is a different field for this anyway, right?

> Prevent commands being processed by the wrong file-writer instance.
> Prevent bad commands.

I think you just mean "make it really awkward for anything other than your Python library to produce the commands" and I very much disagree that that is a good idea.

> Add some extra meta-data for debugging.

Surely explicit fields for such information would be preferable.

Reply:
@zjttoefs any opinion on this?

Reply:
> > Prevent commands being processed by the wrong file-writer instance.
> 
> How? There is a different field for this anyway, right?
> 
I am very confident that the answer to that is: no. I do not mind you correcting me though.

> > Prevent commands being processed by the wrong file-writer instance.
> > Prevent bad commands.
> 
> I think you just mean "make it really awkward for anything other than your Python library to produce the commands" and I very much disagree that that is a good idea. Being able to use different software and scripts to produce the message has been a big advantage of the system that has made it easier to test and to integrate with systems at other institutions.
> 
All correct. It has also caused a lot of headache due to the file-writer not reliably working as expected in anything other than a "laboratory setting" for most of its existence.

> > Add some extra meta-data for debugging.
> 
> Surely explicit fields for such information would be preferable.
I really see no harm in itself in using the identifier for also encoding some meta-data (as long as it is not too much). Also, I do not consider it a good use of my time to push a change to a schema like that through when the existing schema works just fine for what I want to do.